singularity/code/savegame.py

```diff
@@ -687,6 +688,36 @@ def savegame_exists(savegame_name):
 
     return True
 
+def sanitize_filename(filename):
+    """Create filename that's safe for all operating systems"""
+    filename = filename.strip()
+    # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx
+    # http://www.linfo.org/file_name.html
+    # https://kb.acronis.com/content/39790
+
+    # Characters that are allowed in filenames, but not at the beginning
+    filename = re.sub('^[.-]', '_', filename)
+
+    # Characters that are disallowed anywhere in a filename
+    # No potential file separators or other potentially illegal characters
+    filename = re.sub('[<>:"|?*/\\\\]', '_', filename)
+
+    # Replace whitespace
+    filename = re.sub('\s', '_', filename).strip('_')
+
+    # Reserved filenames under Windows
+    windows_reserved = ['CON', 'PRN', 'AUX', 'NUL', 'COM1', 'COM2', 'COM3', 'COM4',
```

Comment:
Based on the usage, please move it out of the function and make it a set a la:

```python
WINDOWS_RESERVED = {'CON', 'PRN', ...}
...

   if filename.upper() in WINDOWS_RESERVED:
```

Reply:
Done